CI: new failures: segfault in free-threaded, `linprog` invalid value encountered in `dot`

### Describe your issue.

CI is failing with a segfault in the free-threaded job; see e.g. gh-22698.

I'm also seeing a failure in the mypy job, e.g. gh-22695.

### Reproducing Code Example

```python
NA
```

### Error message

```shell
In free-threaded(1):
...
scipy/sparse/tests/test_matrix_io.py .......                             [ 99%]
scipy/sparse/tests/test_minmax1d.py ............                         [ 99%]
Fatal Python error: Segmentation fault

Current thread 0x00007f379c41eb80 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/traceback.py", line 200 in _safe_string
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/traceback.py", line 1053 in __init__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/traceback.py", line 1134 in __init__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/traceback.py", line 176 in format_exception_only
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 592 in exconly
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 884 in get_exconly
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 872 in get_source
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 938 in repr_traceback_entry
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 992 in repr_traceback
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 1058 in repr_excinfo
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/_code/code.py", line 690 in getrepr
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/nodes.py", line 456 in _repr_failure_py
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/python.py", line 1669 in repr_failure
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/reports.py", line 376 in from_item_and_call
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/runner.py", line 368 in pytest_runtest_makereport
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/runner.py", line 244 in call_and_report
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/runner.py", line 132 in runtestprotocol
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/main.py", line 362 in pytest_runtestloop
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/main.py", line 337 in _main
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/home/runner/work/scipy/scipy/build-install/lib/python3.13t/site-packages/scipy/_lib/_testutils.py", line 131 in __call__
  File "/home/runner/work/scipy/scipy/dev.py", line 830 in scipy_tests
  File "/home/runner/work/scipy/scipy/dev.py", line 852 in run
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/action.py", line 461 in execute
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/task.py", line 473 in execute
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/runner.py", line 176 in execute_task
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/runner.py", line 220 in run_tasks
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/runner.py", line 254 in run_all
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/cmd_run.py", line 265 in _execute
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/cmd_base.py", line 570 in execute
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/cmd_base.py", line 150 in parse_execute
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/doit/api.py", line 50 in run_tasks
  File "/home/runner/work/scipy/scipy/dev.py", line 252 in run_doit_task
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/pydevtool/cli.py", line 267 in callback
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/click/core.py", line 788 in invoke
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/click/core.py", line 1443 in invoke
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/click/core.py", line 1697 in invoke
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/rich_click/rich_command.py", line 166 in main
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/click/core.py", line 1161 in __call__
  File "/opt/hostedtoolcache/Python/3.13.2/x64-freethreaded/lib/python3.13t/site-packages/rich_click/rich_command.py", line 378 in __call__
  File "/home/runner/work/scipy/scipy/dev.py", line 1740 in <module>

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, cython.cimports.libc.math, scipy._lib._ccallback_c, scipy._lib._fpumode, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.spatial.transform._rotation, scipy.spatial.transform._rigid_transform, scipy.cluster._vq, scipy.cluster._hierarchy, scipy.cluster._optimal_leaf_ordering, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize._cython_nnls, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.integrate._lsoda, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.interpolate._bspl, scipy.special.cython_special, scipy.stats._stats, scipy.stats._biasedurn, scipy.stats._stats_pythran, scipy.stats._levy_stable.levyst, scipy.stats._ansari_swilk_statistics, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._rcont.rcont, scipy.stats._qmvnt_cy, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, _ni_label, scipy.ndimage._ni_label, scipy.fftpack.convolve, scipy.integrate._test_odeint_banded, scipy.io.matlab._mio_utils, scipy.io.matlab._streams, scipy.io.matlab._mio5_utils, scipy.io._test_fortran, scipy.ndimage._ctest, _cytest, scipy.ndimage._cytest, scipy.odr.__odrpack, scipy.optimize.cython_optimize._zeros, scipy.signal._sigtools, scipy.signal._max_len_seq_inner, scipy.signal._upfirdn_apply, scipy.signal._spline, scipy.signal._sosfilt, scipy.signal._peak_finding_utils, extending, extending_cpp (total: 121)
/home/runner/work/_temp/a8dac759-6f15-4aa4-9faa-9bab9b38bb77.sh: line 3:  9686 Segmentation fault      (core dumped) python dev.py --no-build test $TEST_SUBMODULES -- --parallel-threads=4
scipy/sparse/tests/test_sparsetools.py 

---

In mypy job:
=========================== short test summary info ============================
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexDefault::test_zero_column_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexDefault::test_unbounded_no_nontrivial_constraints_1 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexDefault::test_unbounded_no_nontrivial_constraints_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexDefault::test_enzo_example_c_with_unboundedness - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexBland::test_zero_column_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexBland::test_unbounded_no_nontrivial_constraints_1 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexBland::test_unbounded_no_nontrivial_constraints_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogSimplexBland::test_enzo_example_c_with_unboundedness - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPDense::test_zero_column_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPDense::test_unbounded_no_nontrivial_constraints_1 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPDense::test_unbounded_no_nontrivial_constraints_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPDense::test_enzo_example_c_with_unboundedness - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparse::test_zero_column_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparse::test_unbounded_no_nontrivial_constraints_1 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparse::test_unbounded_no_nontrivial_constraints_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparse::test_enzo_example_c_with_unboundedness - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSCommon::test_zero_column_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSCommon::test_unbounded_no_nontrivial_constraints_1 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSCommon::test_unbounded_no_nontrivial_constraints_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSCommon::test_enzo_example_c_with_unboundedness - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSBland::test_zero_column_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSBland::test_unbounded_no_nontrivial_constraints_1 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSBland::test_unbounded_no_nontrivial_constraints_2 - RuntimeWarning: invalid value encountered in dot
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogRSBland::test_enzo_example_c_with_unboundedness - RuntimeWarning: invalid value encountered in dot
FAILED scipy/sparse/linalg/_isolve/tests/test_gcrotmk.py::TestGCROTMK::test_denormals - RuntimeWarning: invalid value encountered in dot
= 25 failed, 60324 passed, 3970 skipped, 192 xfailed, 13 xpassed in 337.24s (0:05:37) =
```

### SciPy/NumPy/Python version and system information

```shell
NA
```

--- a/.github/workflows/array_api.yml
+++ b/.github/workflows/array_api.yml
@@ -48,7 +48,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: ${{ matrix.python-version }}
         cache: 'pip'  # not using a path to also cache pytorch
@@ -88,7 +88,7 @@ jobs:
         echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
 
     - name: Setup compiler cache
-      uses:  actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+      uses:  actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
       id:    cache-ccache
       with:
         path: ${{ steps.prep-ccache.outputs.dir }}

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -33,7 +33,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: ${{ matrix.python-version }}
 

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -49,7 +49,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: ${{ matrix.python-version }}
         cache: 'pip'
@@ -86,7 +86,7 @@ jobs:
         echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
 
     - name: Setup compiler cache
-      uses:  actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+      uses:  actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
       id:    cache-ccache
       # Reference: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
       # NOTE: The caching strategy is modeled in a way that it will always have a unique cache key for each workflow run
@@ -245,7 +245,7 @@ jobs:
           submodules: recursive
 
       - name: Setup Python
-        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
         with:
           python-version: "3.11"
 
@@ -303,7 +303,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: ${{ matrix.python-version }}
 
@@ -313,7 +313,7 @@ jobs:
         sudo apt-get install -y libgmp-dev libmpfr-dev libmpc-dev ccache gfortran lcov
 
     - name: Caching Python dependencies
-      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+      uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
       id: cache
       with:
         path: ~/.cache/pip
@@ -337,7 +337,7 @@ jobs:
         echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
 
     - name: Setup compiler cache
-      uses:  actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+      uses:  actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
       id:    cache-ccache
       with:
         path: ${{ steps.prep-ccache.outputs.dir }}
@@ -466,7 +466,7 @@ jobs:
       with:
         submodules: recursive
         fetch-tags: true
-    - uses: actions/setup-python@9e62be81b28222addecf85e47571213eb7680449 # v5.5.0-dev
+    - uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: '3.13t'
 
@@ -548,7 +548,7 @@ jobs:
           submodules: recursive
 
       - name: Setup Python
-        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
         with:
           python-version: "3.12"
 
@@ -584,7 +584,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: '3.12'
 

--- a/.github/workflows/linux_intel_oneAPI.yml
+++ b/.github/workflows/linux_intel_oneAPI.yml
@@ -50,7 +50,7 @@ jobs:
 
       - name: cache install
         id: cache-install
-        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
         with:
           path: |
               /opt/intel/oneapi/
@@ -72,7 +72,7 @@ jobs:
           bash tools/install_intel_oneAPI_linux.sh $LINUX_HPCKIT_URL
 
       - name: Setup Conda
-        uses: conda-incubator/setup-miniconda@d2e6a045a86077fb6cad6f5adf368e9076ddaa8d # v3.1.0
+        uses: conda-incubator/setup-miniconda@505e6394dae86d6a5c7fbb6e3fb8938e3e863830 # v3.1.1
         with:
           python-version: ${{ matrix.python-version }}
           channels: conda-forge

--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -57,7 +57,7 @@ jobs:
         echo "timestamp=${NOW}" >> $GITHUB_OUTPUT
 
     - name: Setup compiler cache
-      uses:  actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+      uses:  actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
       id:    cache-ccache
       # Reference: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
       # NOTE: The caching strategy is modeled in a way that it will always have
@@ -75,7 +75,7 @@ jobs:
           ${{ github.workflow }}-${{ matrix.python-version }}-ccache-macos-
 
     - name: Setup Conda
-      uses: conda-incubator/setup-miniconda@d2e6a045a86077fb6cad6f5adf368e9076ddaa8d # v3.1.0
+      uses: conda-incubator/setup-miniconda@505e6394dae86d6a5c7fbb6e3fb8938e3e863830 # v3.1.1
       with:
         python-version: ${{ matrix.python-version }}
         channels: conda-forge
@@ -92,7 +92,7 @@ jobs:
       shell: bash
 
     - name: Cache conda
-      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+      uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
       env:
         # Increase this value to reset cache if environment.yml has not changed
         CACHE_NUMBER: 1
@@ -149,7 +149,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: ${{ matrix.python-version }}
         cache: 'pip'
@@ -194,7 +194,7 @@ jobs:
         submodules: recursive
 
     - name: Setup Python
-      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
       with:
         python-version: ${{ matrix.python-version }}
         cache: 'pip'

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -104,7 +104,7 @@ jobs:
         with:
           submodules: true
 
-      - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      - uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
         with:
           python-version: 3.11
 
@@ -212,7 +212,7 @@ jobs:
           echo "CIBW_BUILD_FRONTEND=$CIBW" >> "$GITHUB_ENV"
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@7940a4c0e76eb2030e473a5f864f291f63ee879b # v2.21.3
+        uses: pypa/cibuildwheel@d04cacbc9866d432033b1d09142936e6a0e2121a # v2.23.2
         env:
           CIBW_BUILD: ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}*
           CIBW_ARCHS: ${{ matrix.buildplat[2] }}
@@ -228,14 +228,14 @@ jobs:
             mv ./wheelhouse/*.whl $(find ./wheelhouse -type f -name '*.whl' | sed 's/13_0/14_0/')
           fi
 
-      - uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+      - uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
         with:
           path: ./wheelhouse/*.whl
           name: ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}
             ${{ matrix.buildplat[2] }} ${{ matrix.buildplat[3] }}
             ${{ matrix.buildplat[4] }}
 
-      - uses: conda-incubator/setup-miniconda@505e6394dae86d6a5c7fbb6e3fb8938e3e863830  # v3.1.1
+      - uses: conda-incubator/setup-miniconda@505e6394dae86d6a5c7fbb6e3fb8938e3e863830 # v3.1.1
         with:
           # for installation of anaconda-client, required for upload to
           # anaconda.org

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -36,7 +36,7 @@ jobs:
         with:
           submodules: recursive
       - name: Setup Python
-        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
         with:
           python-version: '3.12'
           architecture: 'x64'
@@ -77,7 +77,7 @@ jobs:
         with:
           submodules: recursive
       - name: Setup Python
-        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
         with:
           python-version: '3.11'
           cache: 'pip'
@@ -120,7 +120,7 @@ jobs:
         with:
           submodules: recursive
       - name: Setup Python
-        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
         with:
           python-version: '3.11'
           cache: 'pip'

--- a/.github/workflows/windows_intel_oneAPI.yml
+++ b/.github/workflows/windows_intel_oneAPI.yml
@@ -49,7 +49,7 @@ jobs:
 
       - name: cache install
         id: cache-install
-        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
         with:
           path: |
               C:\Program Files (x86)\Intel\oneAPI\compiler
@@ -68,7 +68,7 @@ jobs:
           tools/install_intel_oneAPI_windows.bat %WINDOWS_HPC_URL%
 
       - name: Setup Conda
-        uses: conda-incubator/setup-miniconda@d2e6a045a86077fb6cad6f5adf368e9076ddaa8d # v3.1.0
+        uses: conda-incubator/setup-miniconda@505e6394dae86d6a5c7fbb6e3fb8938e3e863830 # v3.1.1
         with:
           python-version: 3.11
           channels: conda-forge

--- a/scipy/sparse/tests/test_sparsetools.py
+++ b/scipy/sparse/tests/test_sparsetools.py
@@ -21,6 +21,7 @@ def int_to_int8(n):
     return (n + 128) % 256 - 128
 
 
+@pytest.mark.thread_unsafe  # Exception handling in CPython 3.13 has races
 def test_exception():
     assert_raises(MemoryError, _sparsetools.test_throw_error)
 

